### PR TITLE
Icu 10439 refactor layout sidebar into rose layout body content

### PR DESCRIPTION
**Pull request by cameronperera**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/JIRA_TICKET_NUMBER)

## Description

- Move `<Layout::Sidebar />` to `<Rose::Layout::BodyContent />`
- Refactor uses of `<Layout::Sidebar />` to `<Rose::Layout::BodyContent />` in Target Details page, Channels-by-Connection, and Channel Details page
- Remove `<Layout::Sidebar />`

:technologist: [Admin preview](https://boundary-39amllt16-hashicorp.vercel.app/scopes/global/session-recordings/sr_y7c8cv7a0e/channels-by-connection)

:rose: [Rose preview](https://boundary-ui-storybook-lvpz8568u-hashicorp.vercel.app/?path=/story/rose-layout-bodycontent--body-content)

### Screenshots (if appropriate):
**Should see no visible difference**
before:
Screen Shot 2023-07-27 at 12 36 37 PM
after:
Screen Shot 2023-07-27 at 12 37 54 PM

### CODE REVIEW

File: ui/admin/app/styles/app.scss [hunk 1]
```diff
@@ -870,25 +870,9 @@
   }
 }
 
-// Layout - Sidebar
-.layout-sidebar {
```
Comment (calcaide) [inline ui/admin/app/styles/app.scss:874]: I think we want to review this. If we get rid of all these styles, the sidebar isn't responsive anymore.

**Review by calcaide — CHANGES_REQUESTED** [review]
After UI manual testing I find out the responsivenes of the "sidebar" is broken. I attached screenshots of before and after.

Screenshot 2023-07-27 at 11 29 50 AM
Screenshot 2023-07-27 at 11 30 04 AM

**Review by ZedLi — APPROVED** [review]
Looks good!

File: ui/admin/app/styles/app.scss (hunk 1 above)
Comment (cameronperera) [reply ui/admin/app/styles/app.scss:874]: Small screen issue fixed.

**Review by DhariniJeeva — APPROVED** [review]

**Comment by cameronperera** [comment]
Here is what we currently have. The media rule max-width is set to `large` (1024px) which lets the HDS table get really small and causes overflow. 
large-media
Instead of trying to make the HDS table change, I decided to make the max-width for the media rule to `xlarge` (1200px). This tells the styles to move the sidebar down sooner and not smash the body and sidebar before finally applying the smaller screen styles.
xlarge-media

**Review by lisbet-alvarez — APPROVED** [review]

**Pull request merged**